pysat/instruments/nasa_cdaweb_methods.py

```diff
@@ -354,35 +412,74 @@ def list_remote_files(tag, sat_id,
     # naming scheme for files on the CDAWeb server
     format_str = inst_dict['remote_fname']
 
+    # Check for appropriate combination of kwargs.  Warn and continue if not.
+    if (year is None) and (month is not None):
+        warnings.warn("Month keyword requires year.  Ignoring month.")
+        month = None
+    if ((year is None) or (month is None)) and (day is not None):
+        warnings.warn("Day keyword requires year and month.  Ignoring day.")
+        day = None
+
     # get a listing of all files
     # determine if we need to walk directories
 
-    soup = BeautifulSoup(requests.get(remote_url).content, "lxml")
-
-    # Find Subdirectories if needed
+    # Find Subdirectories and modify remote_url if user input is specified
     dir_split = os.path.split(format_str)
-    if (len(dir_split) == 2) & (len(dir_split[0]) != 0):
-        links = soup.find_all('a', href=True)
-        dirs = []
-        for link in links:
-            if link['href'].count('/') == 1:
-                dirs.append(link['href'])
-        # only want to keep file portion of the string
-        format_str = dir_split[-1]
-    elif len(dir_split) == 2:
-        # no extra directories
-        dirs = ['']
+    if len(dir_split[0]) != 0:
+        subdirs = dir_split[0].split('/')  # Get all subdirectories
```

Comment:
I think pep8 frowns upon comments to the side?

Reply:
updated